source/2_platform/rex_directx/src/directx_feature_shader_model.cpp

```diff
@@ -0,0 +1,76 @@
+#include "rex_directx/directx_feature_shader_model.h"
+
+#include "rex_directx/directx_util.h"
+
+namespace rex
+{
+  //-------------------------------------------------------------------------
+  rsl::string shader_model_name(s32 shaderModel)
+  {
+    switch(static_cast<D3D_SHADER_MODEL>(shaderModel))
+    {
+      case D3D_SHADER_MODEL::D3D_SHADER_MODEL_6_6: return rsl::string("D3D_SHADER_MODEL_6_6");
+      case D3D_SHADER_MODEL::D3D_SHADER_MODEL_6_5: return rsl::string("D3D_SHADER_MODEL_6_5");
+      case D3D_SHADER_MODEL::D3D_SHADER_MODEL_6_4: return rsl::string("D3D_SHADER_MODEL_6_4");
+      case D3D_SHADER_MODEL::D3D_SHADER_MODEL_6_2: return rsl::string("D3D_SHADER_MODEL_6_2");
+      case D3D_SHADER_MODEL::D3D_SHADER_MODEL_6_1: return rsl::string("D3D_SHADER_MODEL_6_1");
+      case D3D_SHADER_MODEL::D3D_SHADER_MODEL_6_0: return rsl::string("D3D_SHADER_MODEL_6_0");
+      case D3D_SHADER_MODEL::D3D_SHADER_MODEL_5_1: return rsl::string("D3D_SHADER_MODEL_5_1");
+      default: return rsl::string("Unknown shader model");
+    }
+  }
+
+  //-------------------------------------------------------------------------
+  bool check_for_shader_model_support(ID3D12Device* device, s32 version)
+  {
+    D3D12_FEATURE_DATA_SHADER_MODEL shader_model = {};
+    shader_model.HighestShaderModel              = static_cast<D3D_SHADER_MODEL>(version);
+
+    const HRESULT hr = device->CheckFeatureSupport(D3D12_FEATURE_SHADER_MODEL, &shader_model, sizeof(shader_model));
+
+    return SUCCEEDED(hr);
+  }
+
+  //-------------------------------------------------------------------------
+  ShaderModelInfo query_shader_model_version(ID3D12Device* device)
+  {
+    D3D12_FEATURE_DATA_SHADER_MODEL shader_model = {};
+
+#if defined(NTDDI_WIN10_VB) && (NTDDI_VERSION >= NTDDI_WIN10_VB)
```

Comment:
I believe the expected shader model should be constant and not based on what's available.
this will return different results depending on which machine it's compiled

Reply:
Yes currently there are no settings available so I will keep it like this, I'll add an issue that explain we need some for of engine settings which will store the shader model and the fallback model